src/exec.js

```diff
@@ -15,12 +15,11 @@ common.register('exec', _exec, {notUnix:true, canReceivePipe: true});
 // Node is single-threaded; callbacks and other internal state changes are done in the
 // event loop).
 function execSync(cmd, opts, pipe) {
-  var tempDir = _tempDir();
-  var stdoutFile = path.resolve(tempDir+'/'+common.randomFileName()),
-      stderrFile = path.resolve(tempDir+'/'+common.randomFileName()),
-      codeFile = path.resolve(tempDir+'/'+common.randomFileName()),
-      scriptFile = path.resolve(tempDir+'/'+common.randomFileName()),
-      sleepFile = path.resolve(tempDir+'/'+common.randomFileName());
+  var stdoutFile = mktemp('-u')[0],
+      stderrFile = mktemp('-u')[0],
+      codeFile = mktemp('-u')[0],
+      scriptFile = mktemp('-u')[0],
+      sleepFile = mktemp('-u')[0];
```

Comment:
If you switch to the automated option parsing, you'll need to switch these calls to `mktemp({unsafe: true})`.